nf_core/sync.py

```diff
@@ -85,6 +85,9 @@ def sync(self):
         if self.make_pr:
             log.info("Will attempt to automatically create a pull request")
 
+        if os.environ.get("GITHUB_AUTH_TOKEN", "") == "":
+            raise SyncException("GITHUB_AUTH_TOKEN not set!")
```

Comment:
We need to be a little less brutal than this... You can run `nf-core sync` on your laptop to manually sync a pipeline and it shouldn't try to do anything with GitHub (I do this when helping out people quite frequently - I'm probably the only one though!). So we don't want to halt all execution before do anything, we just want to skip the stuff that makes PRs and comments etc.

This is why I had the two custom exception types (which could maybe be renamed to be clearer now) - `SyncException` was for stuff that went wrong with the local stuff and `PullRequestException` was for anything with GitHub.

Reply:
See also logic in the code below:

https://github.com/nf-core/tools/blob/fdc4e06c9f9b4e2414060f3bde3ad10de3ce301c/nf_core/sync.py#L94-L101

1. We only need `GITHUB_AUTH_TOKEN` when `self.make_pr` is `True`
2. We catch the GitHub errors as elegantly as possible and clean up our mess before exiting (as they are more frequent than other error types)

Reply:
Okay, I'll adapt it

Reply:
Raising a `PullRequestException` within the `if` branch now if the token doesn't exist

Reply:
On my phone so not 100% sure but I think that this won't be sufficient. That exception will still halt execution I think. Check needs to come later in the code after all of the local sync stuff is done and inside an `if` block that checks if the command line option for creating a PR was used (so that if you don't ask for any interaction with GitHub it won't throw an error about missing auth).

Reply:
Ah you did move it, sorry. Ok ace 👌🏻